client.go

```diff
@@ -364,6 +364,28 @@ func (c *Client) SetTagByLocator(locator string, tags *Tags) (*Tags, error) {
 	return tags, nil
 }
 
+func (c *Client) DownloadBuildLog(buildId int, w io.WriteCloser) error {
+	url := c.host + fmt.Sprintf("/downloadBuildLog.html?buildId=%d", buildId)
+	var body io.Reader
+
+	req, err := http.NewRequest("GET", url, body)
+	if err != nil {
+		return err
+	}
+
+	req.SetBasicAuth(c.username, c.password)
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
```

Comment:
Since both of these `Close()` invocations can return errors (as can `io.Copy`), perhaps we should not defer them but close them after the copy, capturing and returning any error encountered across these 3 commands.

Reply:
Done